### DISCUSSION TITLE
LDAP Sync works while LDAP auth failes with the same Settings

### DISCUSSION THREAD

**Discussion by Fabl0s** [modoboa/modoboa#2003] [Help]
# Impacted versions

* Modoboa: 1.15.0
* installer used: Yes
* Webserver: Nginx

# Steps to reproduce

Setup LDAP and LDAP Sync, import users via ldap sync command

# Current behavior

LDAP Sync works as intended - but the user is not able to login

# Expected behavior

LDAP Imported user should be able to login

Also, if I got it right, login attempts should be logged in /var/log/syslog too? There is no auth attempt appearing.
Debugging is enabled in the settings.py.

**Comment by tonioo** [comment]
@Fabl0s Have you enabled LDAP logs on your server?

**Comment by Fabl0s** [comment]
@tonioo I'm not sure where I can do that

**Comment by tonioo** [comment]
@Fabl0s I guess you can play with the ``LOGGING`` setting. See https://django-auth-ldap.readthedocs.io/en/latest/logging.html.

**Comment by Fabl0s** [comment]
OK, first issue was on my end with a missing Variable in the Search Filter.

Now that I fixed it and django finds a matching user, I get this Message on Login:

Fehler: Konnte Passwort nicht aktualisieren: {'msgtype': 103, 'msgid': 3, 'result': 34, 'desc': 'Invalid DN syntax', 'ctrls': [], 'info': "0000208F: NameErr: DSID-0310022E, problem 2006 (BAD_NAME), data 8350, best match of:\n\t'None'\n"}

The Debug message is this:

`Sep 11 20:23:10 Modoboa django_auth_ldap[1351]: DEBUG search_s('OU=Users,OU=domain,DC=domain,DC=tld', 2, '(&(|(sAMAccountName=%(user)s))(|(|(memberof:1.2.840.113556.1.4.1941:=CN=modoboa,OU=ServiceGroups,OU=domain,DC=domain,DC=tld)(primaryGroupID=2603))))') returned 1 objects: cn=vorname nachname,ou=users,ou=domain,dc=domain,dc=tld`

`Sep 11 20:23:10 Modoboa django_auth_ldap[1351]: DEBUG search_s('OU=Groups,OU=domain,DC=domain,DC=tld', 2, '(&(objectClass=group)(member=cn=vorname nachname,ou=users,ou=domain,dc=domain,dc=tld))') returned 1 objects: cn=admin,ou=groups,ou=domain,dc=domain,dc=tld`

`Sep 11 20:23:10 Modoboa django_auth_ldap[1351]: DEBUG Populating Django user vorname.nachname`

**Comment by tonioo** [comment]
@Fabl0s ``'desc': 'Invalid DN syntax'``
Looks like you've got an issue with your DN template...

**Comment by Fabl0s** [comment]
@tonioo If you mean the User DN Template setting in Modoboa... I dont have his option with search and bind.
Should I try with the direct bind?

**Comment by tonioo** [comment]
@Fabl0s Indeed this option is not used in search&bind mode but maybe you can try directbind. What's strange in your logs is that a user and a group are found! I don't see any link between the error your encounter and the logs...

**Comment by Fabl0s** [comment]
@tonioo I copied the Search Filter to the DN Template - Now I get LDAP Server refused DN/Password and nothing else in the log.
I double checked username and password tho and those are fine.
Any Ideas what might go wrong?

(&(|(sAMAccountName=%(user)s))(|(|(memberof:1.2.840.113556.1.4.1941:=CN=modoboa,OU=ServiceGroups,OU=Domain,DC=Domain,DC=Tld)(primaryGroupID=2603))))

**Comment by tonioo** [comment]
@Fabl0s Search filter and DN template can't be the same. Do you have a working example of a DN user?

**Comment by Fabl0s** [comment]
@tonioo If you mean by a working example the DN Path to a working User then yes. If you mean a working string for the DN Template I guess not.

**Comment by tonioo** [comment]
@Fabl0s Yes I mean a working user.

**Comment by Fabl0s** [comment]
@tonioo Yes I do - It's required for Search and Bind.

**Comment by tonioo** [comment]
@Fabl0s Can you give me an example? And also the value you set in the DN template field?

**Comment by Fabl0s** [comment]
@tonioo The Working LDAP User is 
"CN=ldap,OU=ServiceAccounts,OU=Domain,DC=Domain,DC=tld"

And the current DN Template is 
"(&(|(sAMAccountName=%(user)s))(|(|(memberof:1.2.840.113556.1.4.1941:=CN=modoboa,OU=ServiceGroups,OU=Domain,DC=Domain,DC=tld)(primaryGroupID=2603))))"

**Comment by tonioo** [comment]
Ok your template should look like this:
CN=%(user)s,OU=ServiceAccounts,OU=Domain,DC=Domain,DC=tld

But I guess it won't work since your schema looks more complicated.

**Comment by Fabl0s** [comment]
Looks Complicated, but not neccessary I think since they get filtered on LDAP Sync anyway - So only modoboa group members should be able to login anyway?

But even with that DN Template I still get a rejection...
Is it maybe the Password Attribute?

**Comment by tonioo** [comment]
Maybe. Have you changed it?

**Comment by Fabl0s** [comment]
More by accident - it's userPassword at the moment.

**Comment by tonioo** [comment]
Actually I don't think it's the problem here. Are you using openldap or activedirectory?

**Comment by Fabl0s** [comment]
It's an Active Directory

**Comment by tonioo** [comment]
@Fabl0s Unfortunately I don't have any ActiveDirectory to test something...

**Comment by liszca** [comment]
> And the current DN Template is
> "(&(|(sAMAccountName=%(user)s))(|(|(memberof:1.2.840.113556.1.4.1941:=CN=modoboa,OU=ServiceGroups,OU=Domain,DC=Domain,DC=tld)(primaryGroupID=2603))))"

Your search filter looks a little bit wired to me, two things, the colons and the operators. Here is how I would write it, but leaving the colons as is:
`(&(sAMAccountName=%(user)s)(|(memberof:1.2.840.113556.1.4.1941:=CN=modoboa,OU=ServiceGroups,OU=Domain,DC=Domain,DC=tld)(primaryGroupID=2603)))`
